### PR TITLE
 [5/N] Fix clang-tidy warnings in torch/csrc/api/ 

### PR DESCRIPTION
**Pull request by cyyever**
Follows #138382

### CODE REVIEW

File: torch/csrc/api/include/torch/linalg.h [hunk 1]
```diff
@@ -1,10 +1,11 @@
 #pragma once
 
 #include <ATen/ATen.h>
+#include <ATen/Tensor.h>
 
 namespace torch {
 namespace linalg {
-
+using at::Tensor;
```
Comment (Skylion007) [inline torch/csrc/api/include/torch/linalg.h:8]: Why do this? A bit confused.

Comment (cyyever) [reply torch/csrc/api/include/torch/linalg.h:8]: Could be reverted

**Review by ezyang — APPROVED** [review]

**Pull request closed without merging**